app/controllers/users_controller.rb

```diff
@@ -13,16 +22,58 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      flash[:success] = t :welcome
+      log_in @user
+      flash[:success] = t ".welcome"
       redirect_to @user
     else
       render :new
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      redirect_to root_path
+    else
+      if @user.update_attributes user_params
+        flash[:success] = t ".profile_updated"
+        redirect_to @user
+      else
+        render :edit
+      end
+    end
+  end
+
+  def destroy
+    User.find_by(id: params[:id]).destroy
```

Comment:
If user is nil/don't find the user with the params[:id], what will happen? Will destroy method be called?